scripts/get_s3_permissions.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python
+from os import path
+from pathlib import Path
+
+import boto3
+import fire
+
+AWS_ACCOUNT_FOR_ENV = {
+    "dev": "dev",
+    "qa": "test",
+    "ref": "test",
+    "int": "test",
+    "prod": "prod",
+}
+
+
+def get_account_id(env: str):
+    if env not in AWS_ACCOUNT_FOR_ENV:
+        raise ValueError(f"Invalid environment: {env}")
+
+    account_name = AWS_ACCOUNT_FOR_ENV[env]
+    secretsmanager = boto3.client("secretsmanager", region_name="eu-west-2")
+    secret_id = f"nhsd-nrlf--mgmt--{account_name}-account-id"
+    result = secretsmanager.get_secret_value(SecretId=secret_id)
+    account_id = result["SecretString"]
+
+    return account_id
+
+
+def get_boto_session_for_account(account_id: str):
+    sts = boto3.client("sts", region_name="eu-west-2")
+    result = sts.assume_role(
+        RoleArn=f"arn:aws:iam::{account_id}:role/terraform",
+        RoleSessionName="get-account-id",
+        DurationSeconds=900,
+    )
+    credentials = result["Credentials"]
+
+    return boto3.Session(
+        aws_access_key_id=credentials["AccessKeyId"],
+        aws_secret_access_key=credentials["SecretAccessKey"],
+        aws_session_token=credentials["SessionToken"],
+    )
+
+
+def get_file_folders(s3_client, bucket_name, prefix=""):
+
+    print("Getting file folders to download...")
+    file_names = []
+    folders = []
+
+    default_kwargs = {"Bucket": bucket_name, "Prefix": prefix}
+    next_token = ""
+
+    while next_token is not None:
+        updated_kwargs = default_kwargs.copy()
+        if next_token != "":
+            updated_kwargs["ContinuationToken"] = next_token
+
+        response = s3_client.list_objects_v2(**updated_kwargs)
+        contents = response.get("Contents")
+
+        for result in contents:
+            key = result.get("Key")
+            if key[-1] == "/":
+                folders.append(key)
+            else:
+                file_names.append(key)
+
+        next_token = response.get("NextContinuationToken")
+
+    return file_names, folders
+
+
+def download_files(s3_client, bucket_name, local_path, file_names, folders):
+    print(f"Downloading {len(file_names)} S3 files to temporary direrectory...")
```

Comment:
There should be a check that you could parse each json file, and if not fail. We could raise a separate ticket for this.

Reply:
I feel like this check would make sense in the pipeline instead of here as this will happen on every release and make things longer but also all files in S3 should be valid JSON

Reply:
I also dont think we should fail if 1 file isnt parseable but all the others are we'd still wanna embed the others right for performance so maybe skip it?

Reply:
Then we should put it as acceptance criteria for the pipeline ticket 👍